sway-core/src/asm_generation/from_ir.rs

```diff
@@ -1058,6 +1058,15 @@ impl<'ir> AsmBuilder<'ir> {
                             &instr_reg,
                             instr_val.get_span(self.context),
                         );
+                        self.bytecode.push(Op {
+                            opcode: either::Either::Left(VirtualOp::ADD(
+                                instr_reg.clone(),
+                                self.stack_base_reg.as_ref().unwrap().clone(),
+                                instr_reg.clone(),
+                            )),
+                            comment: "get offset reg for get_ptr".into(),
```

Comment:
OK, so until now we'd never actually ever run this code and it turned out to be buggy.  An argument to introduce coverage tracking to our tests.